### PR TITLE
Level 2/router

### PR DESCRIPTION
**Pull request by OksanaDenysenko**

### CODE REVIEW

File: back.loc/api/v2/router.php [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+ */
+function addItem($data, $conn): void
+{
+    $value_data = Strip_tags(htmlspecialchars($data['text']));
+
+    $stmt = $conn->prepare("INSERT INTO items (text, checked) VALUES (?, 0)"); // prepared request
+    $stmt->bind_param("s", $value_data);
+
+    if ($stmt->execute()) {
+        $id = $stmt->insert_id;
```
Comment (anton-moroz) [inline back.loc/api/v2/router.php:67]: redundant variable

**Review by anton-moroz — CHANGES_REQUESTED** [review]

File: back.loc/api/v2/db_conect.php [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?php
+require 'config.php'; // database settings file
+
+$config = require 'config.php';
```
[the remark this answers is not in the document]
Comment (OksanaDenysenko) [reply back.loc/api/v2/db_conect.php:4]: corrected

File: back.loc/api/v2/db_config.php [hunk 3]
```diff
@@ -31,7 +32,7 @@
     checked BOOL NOT NULL
 )";
 if ($conn->query($sql) === TRUE) {
```
[the remark this answers is not in the document]
Comment (OksanaDenysenko) [reply back.loc/api/v2/db_config.php:34]: replased with foreach

File: back.loc/api/v2/router.php [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+require_once("db_conect.php");
+
+// receives json
+$json = file_get_contents('php://input');
+$data = json_decode($json, true);
+
+// get data via URL and call a certain function depending on the received data
+$action = $_GET['action'];
+
+switch ($action) {
```
[the remark this answers is not in the document]
Comment (OksanaDenysenko) [reply back.loc/api/v2/router.php:25]: replased

File: back.loc/api/v2/router.php (hunk 1 above)
Comment (OksanaDenysenko) [reply back.loc/api/v2/router.php:67]: deleted

File: back.loc/api/v2/router.php [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+    if (mysqli_num_rows($result) != 1) { //if the user is not registered
+        http_response_code(401);
+        exit(json_encode(['error' => 'User not found']));
+    }
+
+    $row = $result->fetch_assoc(); // convert the string into an associative array
+
+    if (!password_verify($password, $row['pass'])) { //if the password hash does not match
+        http_response_code(400);
+        exit(json_encode(['error' => 'Incorrect password']));
```
[the remark this answers is not in the document]
Comment (OksanaDenysenko) [reply back.loc/api/v2/router.php:183]: added

**Review by anton-moroz — CHANGES_REQUESTED** [review]

File: back.loc/api/v2/router.php [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+require_once("db_conect.php");
+
+// receives json
+$json = file_get_contents('php://input');
+$data = json_decode($json, true);
+
+// get data via URL and call a certain function depending on the received data
+$action = $_GET['action'];
+
+match ($action) {
```
[the remark this answers is not in the document]
Comment (OksanaDenysenko) [reply back.loc/api/v2/router.php:28]: Thanks!

File: back.loc/api/v2/db_conect.php [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+$config = require 'config.php'; // database settings file
+
+$conn = new mysqli(
+    $config['dbhost'],
+    $config['dbuser'],
+    $config['dbpass'],
+    $config['dbname']
+);
+
+
```
[the remark this answers is not in the document]
Comment (OksanaDenysenko) [reply back.loc/api/v2/db_conect.php]: deleted

**Review by anton-moroz — APPROVED** [review]

**Pull request closed without merging**